auth_test.py

```diff
@@ -845,29 +845,75 @@ def test_grant_revoke_cleanup(self):
 
         assert_unauthorized(cathy, "SELECT * FROM ks.cf", "User cathy has no SELECT permission on <table ks.cf> or any of its parents")
 
-    def test_permissions_caching(self):
+    def test_permissions_caching_authenticated_user(self):
         """
-        * Launch a one node cluster, with a 2s permission cache
+        This test is to show that the permissions caching in AuthenticatedUser
+        works correctly and revokes the permissions from a logged in user
+        * Launch a one node cluster, with a permissions cache of 2s
         * Connect as the default superuser
-        * Create a new user, 'cathy'
-        * Create a table, ks.cf
-        * Connect as cathy in two separate sessions
-        * Grant SELECT to cathy
-        * Verify that reading from ks.cf throws Unauthorized until the cache expires
-        * Verify that after the cache expires, we can eventually read with both sessions
+        * Create USER cathy
+        * Grant SELECT permissions to cathy
+        * Verify cathy can perform expected operations
+        * Revoke SELECT permissions from cathy.
+        * Try reading as cathy, and verify that eventually the cache expires and it fails.
 
-        @jira_ticket CASSANDRA-8194
+        @jira_ticket STAR-590
         """
         self.prepare(permissions_validity=2000)
 
         cassandra = self.get_session(user='cassandra', password='cassandra')
-        cassandra.execute("CREATE USER cathy WITH PASSWORD '12345'")
         cassandra.execute("CREATE KEYSPACE ks WITH replication = {'class':'SimpleStrategy', 'replication_factor':1}")
         cassandra.execute("CREATE TABLE ks.cf (id int primary key, val int)")
+        cassandra.execute("INSERT INTO ks.cf (id, val) VALUES (0, 0)")
+
+        # grant SELECT to cathy
+        cassandra.execute("CREATE USER cathy WITH PASSWORD '12345'")
+        cassandra.execute("GRANT SELECT ON ks.cf TO cathy")
 
         cathy = self.get_session(user='cathy', password='12345')
+
+        assert_one(cathy, "SELECT * FROM ks.cf", [0, 0])
+
+        # revoke SELECT from cathy
+        cassandra.execute("REVOKE SELECT ON ks.cf FROM cathy")
+
+        # cathy should retain permissions until the cache expires
+        # cache entries are invalidated on REVOKE in 4.0 @jira_ticket STAR-590
+        max_attempts = 1 if self.cluster.version() >= LooseVersion('4.0') else 20
```

Comment:
I am wonder if `20` should be stored in a constant and shared between tests. It is also used in `test_role_caching_authenticated_user`.

Reply:
Maybe, however one is used for Roles cache and one for Permissions cache so it's possible they could also be different. 
Also, there are so many hardcoded, repeated, values being used in these tests that it almost feels like a convention not to do it :)  Python doesn't really have true constants anyway?

Reply:
Thank you for explaining! I am also new to Python :) Resolving.